### PR TITLE
[Review] Request from '' @ 'yast/ycp-killer/messaging-refactoring'

### PR DESCRIPTION
**Pull request by dmajda**
Please review the following changes:
- c583a17 Refactor the Messages module and its usage
- 2d931b9 Use the Messages module for all messages to users
- 6881a71 Introduce Command#file_action
- 63e5237 Wrap everything that can fail in commands into an |action|

### CODE REVIEW

File: lib/messages.rb [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
-      spaces = if @last_message_size < MAX_MESSAGE_WIDTH
-        " " * (MAX_MESSAGE_WIDTH - @last_message_size)
-      else
-        ""
-      end
+    def item(message)
+      puts "  * #{message}..."
 
-      puts spaces + status
+      @prefix = "    "
```
Comment (jreidinger) [inline lib/messages.rb:18]: this prefix handling seem quite hackish for me.
What about state machine and have method prefix, that return indentation depending on state?
like

```
@state = :item
def prefix
  case @state
    when :item then " "*4
    when :header then " "*2
    when :default then ""
    default raise "Unknown value"
  end
```

of course better state names would help imagine what the state actually is.

File: yk [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env ruby
 
 require "cheetah"
+require "smart_colored/extend"
```
Comment (jreidinger) [inline yk:4]: you don't use it in this file, so no need to require it.

File: yk [hunk 3]
```diff
@@ -218,10 +221,10 @@ class CLI < Thor
     max_counter_size = 3 + ((Math.log10(modules.size).to_i + 1) * 2)
 
     modules.each_index do |i|
-      printf(
+      Messages.header sprintf(
```
Comment (jreidinger) [inline yk:224]: When I see this code and related to quite stupid reset in Messages I think you should consider make header accepting block, so you can write 

```
message = sprintf ...
Messages.header(message) do
  yield modules[i]
end
```

this code should ensure that after finish, it is reset

File: yk (hunk 2 above)
Comment (jreidinger) [inline yk:4]: ah, my fault, I overlook red at the end. you can ignore it

File: lib/messages.rb (hunk 1 above)
Comment (dmajda) [inline lib/messages.rb:18]: I don't see it as hackish, but I don't mind the state machine either. Fixed in #324.

File: yk (hunk 3 above)
Comment (dmajda) [inline yk:224]: Good idea. I did this both for `Messages.header` and `Messages.item` in #324.

**Comment by dmajda** [comment]
Obsoleted by #324.

**Pull request closed without merging**